src/mavlink_receive.c

```diff
@@ -212,10 +213,9 @@ static void handle_mavlink_message(void)
 // function definitions
 void mavlink_receive(void)
 {
-  while (serialTotalBytesWaiting(Serial1))
+  while (serial_bytes_available())
```

Comment:
I think that could potentially be good; it'd let us support DSM receivers and I don't think it'd be too much feature creep. Of course, that could be handled on the board support side too maybe and be transparent to the flight stack. I think this is something we can leave as-is for now, but will probably want to revisit. The whole RC interface could probably use some cleaning up in general anyway